### DISCUSSION TITLE
tailwind does not work in CRA

### DISCUSSION THREAD

**Discussion by Chandler-Zhu** [tailwindlabs/tailwindcss#9251] [Help] [answered]
I tried to add [this section](https://tailwindcomponents.com/component/tailwind-css-contact-form) at the bottom of my react SPA, but it looks like tailwind didn't kick in, I have already tried `npm install postcss@latest` but still getting the same result

repro link:  [codesandbox](https://codesandbox.io/p/github/Chandler-Zhu/react-parallax/draft/autumn-haze?file=%2Fsrc%2Fcomponents%2FContactTW.jsx&selection=%5B%7B%22endColumn%22%3A7%2C%22endLineNumber%22%3A8%2C%22startColumn%22%3A7%2C%22startLineNumber%22%3A8%7D%5D&workspace=%257B%2522activeFileId%2522%253A%2522cl7n0dsjm000glogc99fdf0mn%2522%252C%2522openFiles%2522%253A%255B%255D%252C%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522gitSidebarPanel%2522%253A%2522COMMIT%2522%252C%2522sidekickItems%2522%253A%255B%257B%2522type%2522%253A%2522PREVIEW%2522%252C%2522taskId%2522%253A%2522start%2522%252C%2522port%2522%253A3000%252C%2522key%2522%253A%2522cl7n0fiiu00802a69hx83vuzb%2522%252C%2522isMinimized%2522%253Afalse%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522start%2522%252C%2522key%2522%253A%2522cl7n0fgno005z2a69qwwep7s4%2522%252C%2522isMinimized%2522%253Afalse%257D%255D%257D)
![image](https://user-images.githubusercontent.com/61914365/188302280-31dec8af-956b-4a1f-b886-5cdcdfb32ed1.png)

**Comment by wongjn** [comment, marked as answer]
[You're using `react-scripts@4.0.3`](https://github.com/Chandler-Zhu/react-parallax/blob/2dadc990f86ab588f060f8bd2421327cc86d69e6/package-lock.json#L13872-L13873) and we can see that in its Webpack configuration, [it has a section for PostCSS](https://github.com/facebook/create-react-app/blob/f92c37a6e6636ed1650ffd976c6881f59b1be803/packages/react-scripts/config/webpack.config.js#L125-L149). `v4.0.3` is using [`postcss-loader@3.0.0`](https://github.com/Chandler-Zhu/react-parallax/blob/2dadc990f86ab588f060f8bd2421327cc86d69e6/package-lock.json#L12036-L12037) and we can see in its source code that [it will use options passed it via loader config if it exists](https://github.com/webpack-contrib/postcss-loader/blob/7647ac9051a17a3a6add7364828bf7bd9955b968/src/index.js#L26-L51), ignoring any `postcss.config.js` file. So your `postcss.config.js` is never read and the Tailwind plugin is never added to the PostCSS processing pipeline

Contrast this with if we used `react-scripts@^5`: [the section for PostCSS takes Tailwind into account](https://github.com/facebook/create-react-app/blob/19fa58d527ae74f2b6baa0867463eea1d290f9a5/packages/react-scripts/config/webpack.config.js#L144-L173), and even if it didn't, it is using [`postcss-loader@^6.2.1`](https://github.com/facebook/create-react-app/blob/19fa58d527ae74f2b6baa0867463eea1d290f9a5/packages/react-scripts/package.json#L59) which [merges any `postcss.config.js` file options with the loader options](https://github.com/webpack-contrib/postcss-loader/tree/v6.2.1#config).

TLDR: Update to `react-scripts@^5` for Tailwind to work.
